### DISCUSSION TITLE
Empfehlung für den Fehler "cc1plus: error: bad value (‘34kc’) for ‘-mtune=’ switch"

### DISCUSSION THREAD

**Discussion by SvenLuebke** [Freetz-NG/freetz-ng#829] [Allgemein]
Hallo allerseits!

Ich wollte freetz-ng mal wieder durchbauen und stoße dabei auf ein "neues" Problem...oder ich hab den Fix schon wieder vergessen:

```
---> toolchain/target/gcc/target ...
building ... PATH="/media/projects/fritzbox/freetz-ng/toolchain/build/mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/mips-linux-uclibc/bin:/media/projects/fritzbox/freetz-ng/toolchain/build/mips_gcc-5.5.0/mips-unknown-linux-gnu/bin:/media/projects/fritzbox/freetz-ng/toolchain/build/mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/mips-linux-uclibc/bin-ccache:/media/projects/fritzbox/freetz-ng/toolchain/build/mips_gcc-5.5.0/mips-unknown-linux-gnu/bin-ccache:/media/projects/fritzbox/freetz-ng/tools/path:/home/kmeleon/perl5/bin:/home/kmeleon/perl5/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/usr/sbin" LD_RUN_PATH="/var/media/ftp/usr/lib" FREETZ_LIBRARY_DIR="/var/media/ftp/usr/lib" make -j4 MAKEINFO=true gcc_cv_libc_provides_ssp=yes -C /media/projects/fritzbox/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/gcc-13.2.0-target all

...und dann ein paar Zeilen später...

g++  -I/media/projects/fritzbox/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/gcc-13.2.0/libcpp -I. -I/media/projects/fritzbox/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/gcc-13.2.0/libcpp/../include -I/media/projects/fritzbox/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/gcc-13.2.0/libcpp/include  -march=34kc -mtune=34kc -msoft-float -O3 -Wa,--trap -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wl,-I/var/media/ftp/usr/lib/ld-uClibc.so.1     -W -Wall -Wno-narrowing -Wwrite-strings -Wmissing-format-attribute -pedantic -Wno-long-long  -fno-exceptions -fno-rtti -I/media/projects/fritzbox/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/gcc-13.2.0/libcpp -I. -I/media/projects/fritzbox/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/gcc-13.2.0/libcpp/../include -I/media/projects/fritzbox/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/gcc-13.2.0/libcpp/include    -c -o charset.o -MT charset.o -MMD -MP -MF .deps/charset.Tpo /media/projects/fritzbox/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-3.10/gcc-13.2.0/libcpp/charset.cc
cc1plus: error: bad value (‘34kc’) for ‘-march=’ switch
cc1plus: note: valid arguments to ‘-march=’ switch are: nocona core2 nehalem corei7 westmere sandybridge corei7-avx ivybridge core-avx-i haswell core-avx2 broadwell skylake skylake-avx512 cannonlake icelake-client icelake-server cascadelake tigerlake cooperlake bonnell atom silvermont slm goldmont goldmont-plus tremont knl knm x86-64 x86-64-v2 x86-64-v3 x86-64-v4 eden-x2 nano nano-1000 nano-2000 nano-3000 nano-x2 eden-x4 nano-x4 k8 k8-sse3 opteron opteron-sse3 athlon64 athlon64-sse3 athlon-fx amdfam10 barcelona bdver1 bdver2 bdver3 bdver4 znver1 znver2 btver1 btver2 native
cc1plus: error: bad value (‘34kc’) for ‘-mtune=’ switch
cc1plus: note: valid arguments to ‘-mtune=’ switch are: nocona core2 nehalem corei7 westmere sandybridge corei7-avx ivybridge core-avx-i haswell core-avx2 broadwell skylake skylake-avx512 cannonlake icelake-client icelake-server cascadelake tigerlake cooperlake bonnell atom silvermont slm goldmont goldmont-plus tremont knl knm intel x86-64 eden-x2 nano nano-1000 nano-2000 nano-3000 nano-x2 eden-x4 nano-x4 k8 k8-sse3 opteron opteron-sse3 athlon64 athlon64-sse3 athlon-fx amdfam10 barcelona bdver1 bdver2 bdver3 bdver4 znver1 znver2 btver1 btver2 generic native
make[4]: *** [Makefile:227: charset.o] Error 1
make[3]: *** [Makefile:3060: all-build-libcpp] Error 2
make[3]: *** Waiting for unfinished jobs....
```
Es wird also etwas mit dem normalen g++ durchgebaut, nicht mit dem g++ der Fritzbox. Klingt aber eher so, als würden die falschen CPPFLAGS verwendet werden, weil für den Host gebaut wird, oder?

Habe [einen Forumeintrag](https://www.ip-phone-forum.de/threads/freetz-ng-fehler-beim-bauen-f%C3%BCr-unterschiedliche-konfigurationen.308030/page-3) gefunden, bei dem das Problem auch aufgetreten ist. Der auf der letzten Seite gepostete Fix https://github.com/dirk-dhu/freetz-ng-new/commit/7fb75639df9838dd9ebdd15c83d2ad10719c7add hat nichts geändert. Seitdem sind aber auch ein paar Sachen umgestellt wurden. Vielleicht hab ich nicht alles korrekt gemerged.

Danke Euch für die Hilfe!

**Comment by fda77** [comment]
Was baust du da für was womit überhaupt? `Build binutils and gcc for target` gibts in NG nicht, das ist eine DHU option. Er hat aber noch einiges andere geändert. Dieser commit müsste noch funkionieren, aber nur mir configure. ka ob dhu das auch für die anderen neuen system adaptiert hat

**Reply by SvenLuebke** [reply]
Eigentlich wollte ich nur ein paar Binaries für meine 7490 unter Freetz-NG bauen und diese aufs Target packen. Dein Kommentar bringt mich schon ein wenig durcheinander :-). Dachte erst, ein "git pull" hätte irgendwann mal nicht richtig funktioniert, aber ich sehe das hier: https://github.com/Freetz-NG/freetz-ng/blob/de4656d16da25deae9b973417c2476e3b9ae73a4/config/ui/toolchain.in#L953

![BuildForTarget](https://github.com/Freetz-NG/freetz-ng/assets/865423/984c9817-884a-436d-89ca-601489526db7)

Eigentlich brauche ich diese Option gar nicht mehr. Kann sein, dass ich perl+gcc zum Update der Module mal installieren wollte. Oder ich habe diese Option falsch verstanden, da das ja anscheinend der gcc für das Target und nicht für den Host ist.

Aber könntest Du vielleicht erklären, warum Du der Meinung bist, dass das nicht mehr auftaucht?

Danke Dir

**Comment by fda77** [comment]
Du hats doch mit den DHU commits angefangen. Bitte dort nachfragen wo du ausgecheckt hast!

**Comment by SvenLuebke** [comment]
> Du hats doch mit den DHU commits angefangen. Bitte dort nachfragen wo du ausgecheckt hast!

Ich nutze natürlich die ganz normalen freetz-NG commits aus diesem Repo, sonst hätte ich natürlich woanders gefragt.

Als ich den Fehler bekommen habe, hab ich nach einer Lösung gesucht. Es gab ein Forumeintrag mit ähnlichem Problem (suehe oben), in welchem als Lösung ein Commit anscheinend aus dem von Dir erwähnten DHU stammt. DHU sagte mir vorher nichts.

```
user@er00186v:/media/projects/fritzbox/freetz-ng$ git remote -v
origin  https://github.com/Freetz-NG/freetz-ng.git (fetch)
origin  https://github.com/Freetz-NG/freetz-ng.git (push)
```

**Reply by fda77** [reply]
Wenn dirk/dhu was schreibt muss man darauf achten über welchen fork er schreibt. Freetz nutzt für manche geräte+fos kombinationen eine von avm unabgängige gcc version. evtl klappt es deshlab nicht
